main.go

```diff
@@ -1,68 +1,160 @@
 package main
 
 import (
-	"os"
-	"fmt"
-	"net"
-	"strings"
+	"bufio"
+	"flag"
+	"io"
+	"log"
 	"math/rand"
-	"io/ioutil"
+	"net"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
 )
 
-func randomQuote() ([]byte, error) {
-	bs, err := ioutil.ReadFile(os.Args[2])
+type quoteServer struct {
+	quotes     []string
+	randPool   sync.Pool // sync.Pool<*rand.Rand>
+	inShutdown bool
+	sync.RWMutex
+	sync.WaitGroup
+}
+
+func newServer(r io.Reader) (*quoteServer, error) {
+	var q quoteServer
+	q.randPool.New = func() interface{} {
+		return rand.New(rand.NewSource(time.Now().UnixNano()))
+	}
+	err := q.load(r)
 	if err != nil {
-		fmt.Println(err)
 		return nil, err
 	}
+	return &q, nil
+}
 
-	q := strings.Split(string(bs), "\n")
-	r := rand.Intn(len(q) - 1)
-
-	return []byte(q[r]), nil
+func (q *quoteServer) load(r io.Reader) error {
+	q.Lock()
+	defer q.Unlock()
+	s := bufio.NewScanner(r)
```

Comment:
Using buffo.Scanner is typically more efficient compared with strings.Split as it does not require holding everything in memory.